rita_common/src/peer_listener/mod.rs

```diff
@@ -161,21 +164,34 @@ fn listen_to_available_ifaces(peer_listener: &mut PeerListener) {
 }
 
 /// Returns a copy of PL lazy static var
-fn get_pl_copy() -> PeerListener {
-    PEER_LISTENER.read().unwrap().clone()
+fn get_pl_copy(key: String) -> Option<PeerListener> {
+    let pl = PEER_LISTENER.read().unwrap().clone();
+    let tmp = pl.get(&key);
+    tmp.cloned()
 }
 
 /// Sets the PeerListener lazy static to the given value
-fn set_pl(pl: PeerListener) {
-    *PEER_LISTENER.write().unwrap() = pl;
+fn set_pl(key: String, pl: PeerListener) {
+    let mut map = PEER_LISTENER.read().unwrap().clone();
+    map.insert(key, pl);
+    *PEER_LISTENER.write().unwrap() = map;
 }
 
 /// Ticks the peer listener module sending ImHere messages and receiving Hello messages from all
 /// peers over UDP
 pub fn peerlistener_tick() -> PeerListener {
     trace!("Starting PeerListener tick!");
 
-    let mut pl = get_pl_copy();
+    let key = if cfg!(feature = "integration_test_v2") {
+        KI.get_namespace().unwrap()
+    } else {
+        "default".to_string()
+    };
```

Comment:
the namespace handling should be abstracted into the get_pl_copy function. Also consider changing the pl functions from a total getter and setter to more operation specific stuff. Take a look at what's been done in the payment tracker module. For example a remove interface function would grab a write lock, remove the interface, then return. This sort of function structure of atomic changes avoids the risk of having a getter/setter pattern where another thread may make and save changes between when this thread gets the data from the lock and sets the version with it's own changes.